### PR TITLE
[Rest Server] Support GPU type in job submission

### PR DESCRIPTION
**Pull request by abuccts**
Support GPU type in job submission.

### CODE REVIEW

File: job-tutorial/README.md [hunk 1]
```diff
@@ -104,6 +105,7 @@ Below please find the detailed explanation for each of the parameters in the con
 | `taskRole.memoryMB`            | Integer, required          | Memory for one task in the task role, no less than 100 |
 | `taskRole.gpuNumber`           | Integer, required          | GPU number for one task in the task role, no less than 0 |
 | `taskRole.command`             | String, required           | Executable command for tasks in the task role, can not be empty |
+| `gpuType`                      | String, optional           | Specify the GPU type to be used in the tasks |
```
Comment (fanyangCS) [inline job-tutorial/README.md]: >tasks [](start = 103, length = 5)

If omitted, the job will run on any gpu type

File: rest-server/src/config/job.js [hunk 2]
```diff
@@ -58,6 +58,9 @@ const jobConfigSchema = Joi.object().keys({
       }))
     .min(1)
     .required(),
+  gpuType: Joi.string()
+    .allow('')
+    .default(''),
```
Comment (fanyangCS) [inline rest-server/src/config/job.js]: >'' [](start = 13, length = 2)

please check with QIngcha about the behavior when gpuType is '' I remember he says empty string will result in non of the gpu type being matched.

File: rest-server/src/models/job.js [hunk 3]
```diff
@@ -277,13 +277,15 @@ class Job {
   }
 
   generateFrameworkDescription(data) {
+    const gpuType = data.gpuType || null;
```
Comment (hwuu) [inline rest-server/src/models/job.js]: Does this line mean that gpuType is null when data.gpuType is ‘’?

Comment (abuccts) [reply rest-server/src/models/job.js]: yes, it's the format required by launcher

**Pull request merged**